app/models/species.rb

```diff
@@ -20,4 +22,31 @@ class Species < ApplicationRecord
   scope :near_threatened, -> { where(redlist_status: NEAR_THREATENED) }
   scope :threatened_and_near, -> { where(redlist_status: [THREATENED, NEAR_THREATENED].flatten) }
   scope :not_threatened, -> { where.not(redlist_status: THREATENED) }
+
+  def self.count_species(species = nil)
+    species ||= all
+    hash = count_by_category_and_habitat(species).compact
+    hash.each do |habitat|
```

Comment:
Is it possible to use a |key, value| (perhaps something like |habitat_id, count_by_category| or similar) syntax here rather than habitat.first and habitat.last? The code like this makes hash look like an array of arrays. Maybe this is just me not being very familiar with ruby conventions!